src/platforms/dummy/dummyiaphandler.cpp

```diff
@@ -18,10 +20,17 @@ DummyIAPHandler::DummyIAPHandler(QObject* parent) : IAPHandler(parent) {
 
 DummyIAPHandler::~DummyIAPHandler() { MVPN_COUNT_DTOR(DummyIAPHandler); }
 
-void DummyIAPHandler::nativeRegisterProducts() {}
+void DummyIAPHandler::nativeRegisterProducts() {
+  QTimer::singleShot(200, this,
+                     [this]() { emit productsRegistrationCompleted(); });
+}
 
 void DummyIAPHandler::nativeStartSubscription(Product* product) {
-  Q_UNUSED(product)
+  Q_ASSERT(product->m_name == "web");
+  // This is a hack. Should we make a new state?
+  MozillaVPN::instance()->setUserState(MozillaVPN::UserNotAuthenticated);
```

Comment:
This is bad, because technically, the user is already authenticated.
I think you need to call the `AuthenticationListener::create() + start()` in a task.